mobile-ui/src/main/java/com/playone/mobile/ui/injection/module/BrowseActivityModule.kt

```diff
@@ -15,6 +33,77 @@ import dagger.Provides
 @Module
 open class BrowseActivityModule {
 
+    @Provides
+    internal fun provideGetBufferoos(bufferooRepository: BufferooRepository,
+                                     threadExecutor: ThreadExecutor,
+                                     postExecutionThread: PostExecutionThread)
+            = GetBufferoos(bufferooRepository, threadExecutor, postExecutionThread)
+
+    @Provides
+    internal fun provideBufferooEntityMapper() = BufferooEntityMapper()
+
+    @Provides
+    internal fun provideBufferooMapper() = com.playone.mobile.presentation.mapper.BufferooMapper()
+
+    @Provides
+    internal fun provideDbBufferooMapper() = com.playone.mobile.cache.db.mapper.BufferooMapper()
+
+    @Provides
+    internal fun provideRemoteBufferooMapper()
+            = com.playone.mobile.remote.mapper.BufferooEntityMapper()
+
+    @Provides
+    internal fun provideDataBufferooMapper() = com.playone.mobile.data.mapper.BufferooMapper()
+
+    @Provides
+    internal fun provideUiBufferooMapper() = com.playone.mobile.ui.mapper.BufferooMapper()
+
+    @Provides
+    internal fun provideBufferooCache(factory: DbOpenHelper,
+                                      entityMapper: BufferooEntityMapper,
+                                      mapper: com.playone.mobile.cache.db.mapper.BufferooMapper,
+                                      helper: PreferencesHelper): BufferooCache
+            = BufferooCacheImpl(factory, entityMapper, mapper, helper)
+
+    @Provides
+    internal fun provideBufferooRemote(
+            service: BufferooService,
+            factory: com.playone.mobile.remote.mapper.BufferooEntityMapper): BufferooRemote
+            = BufferooRemoteImpl(service, factory)
+
+    @Provides
+    internal fun provideBufferooDataStoreFactory(bufferooCache: BufferooCache,
+                                                 bufferooCacheDataStore: BufferooCacheDataStore,
+                                                 bufferooRemoteDataStore: BufferooRemoteDataStore)
+            : BufferooDataStoreFactory
+            = BufferooDataStoreFactory(
+            bufferooCache,
+            bufferooCacheDataStore,
+            bufferooRemoteDataStore)
+
+    @Provides
+    internal fun provideBufferooCacheDataStore(bufferooCache: BufferooCache)
+            : BufferooCacheDataStore
+            = BufferooCacheDataStore(bufferooCache)
```

Comment:
We might not decide this coding style yet.

```kotlin
// 1
fun foo(xxx: Int)
    : Hoo = Hoo(xxx)

// or

// 2
fun foo(xxx: Int)
    : Hoo =
     Hoo(xxx)
```

Which one is better?

Reply:
I prefer the first one. Ofcz, the second line is too long, it will be the second option.

Reply:
I think the same.

Reply:
```
// 0
fun foo(xxx: Int) : Hoo = Hoo(xxx)

If 0 is too long
// 1
fun foo(xxx: Int)
    : Hoo = Hoo(xxx)

If 1 is too long
// 2
fun foo(xxx: Int)
    : Hoo =
     Hoo(xxx)
```